include/cling/Utils/Casting.h

```diff
@@ -0,0 +1,56 @@
+//--------------------------------------------------------------------*- C++ -*-
+// CLING - the C++ LLVM-based InterpreterG :)
+// author: Roman Zulak
+//
+// This file is dual-licensed: you can choose to license it under the University
+// of Illinois Open Source License or the GNU Lesser General Public License. See
+// LICENSE.TXT for details.
+//------------------------------------------------------------------------------
+
+#ifndef CLING_UTILS_CASTING_H
+#define CLING_UTILS_CASTING_H
+
+namespace cling {
+  namespace utils {
+
+    ///\brief Legally cast a pointer to a function to void*.
+    ///
+    /// \param [in] The function to cast
+    ///
+    /// \return A void* of the functions address.
+    ///
+    template <class T>
+    void* FunctionToVoid(T *fPtr) {
+      union { T *f; void *v; } tmp;
+      tmp.f = fPtr;
+      return tmp.v;
+    }
+
+    ///\brief Legally cast a uintptr_t to a function.
+    ///
+    /// \param [in] The uintptr_t to cast
+    ///
+    /// \return The function's address.
+    ///
+    template <class T>
+    T UIntToFunction(uintptr_t ptr) {
+      union { T f; uintptr_t v; } tmp;
+      tmp.v = ptr;
+      return tmp.f;
+    }
+
+    ///\brief Legally cast a void* to a function.
+    ///
+    /// \param [in] The void* to cast
+    ///
+    /// \return The function's address.
+    ///
+    template <class T>
+    T VoidToFunction(void *ptr) {
+      return UintToFunction<T>(static_cast<void*>(ptr));
```

Comment:
Why do you need to `static_cast` a `void*` to a `void*`? Did you mean `static_cast<uintptr_t>`?